src/validationinterface.cpp

```diff
@@ -139,6 +139,8 @@ void UnregisterSharedValidationInterface(std::shared_ptr<CValidationInterface> c
 
 void UnregisterValidationInterface(CValidationInterface* callbacks)
 {
+    callbacks->ValidationInterfaceUnregistering();
```

Comment:
Is it worth to be more specific in the naming whether the unregistering will happen or has already happened?

Like renaming the function to `::ValidationInterfaceWillUnregister`?

This makes it clearer that the callback hasn't been removed yet.

Comparing to introducing a callback which gets called after the fact of the callback being removed `::ValidationInterfaceDidUnregister'